### PR TITLE
Convert all headesr to camel case before merge

### PR DESCRIPTION
**Pull request by windkit**
## Related Issue
https://github.com/leo-project/leofs/issues/489

### CODE REVIEW

File: src/cowboy_req.erl [hunk 1]
```diff
@@ -1222,26 +1222,43 @@ response_merge_headers(Headers, RespHeaders, DefaultHeaders) ->
...
+
 %% Merge headers by prepending the tuples in the second list to the
 %% first list. It also handles Set-Cookie properly, which supports
 %% duplicated entries. Notice that, while the RFC2109 does allow more
 %% than one cookie to be set per Set-Cookie header, we are following
 %% the implementation of common web servers and applications which
 %% return many distinct headers per each Set-Cookie entry to avoid
 %% issues with clients/browser which may not support it.
-merge_headers(Headers, []) ->
+merge_headers_1(Headers, []) ->
```
Comment (mocchira) [inline src/cowboy_req.erl:1254]: Drop a comment why converting Camel Case is needed.
Adding a link to the issue on github would be enough to me.
https://github.com/leo-project/leofs/issues/489

File: src/cowboy_req.erl [hunk 2]
```diff
@@ -1222,26 +1222,43 @@ response_merge_headers(Headers, RespHeaders, DefaultHeaders) ->
 		merge_headers(Headers2, RespHeaders),
 		DefaultHeaders).
 
+camelize_header({N,V}) ->
```
Comment (mocchira) [inline src/cowboy_req.erl]: Matching at function argument is more idiomatic than one at case expression in erlang if possible.
```erlang
camelize_header({<<"ETag">> = N,V}) ->
    {N, V};
camelize_header({<<"x-amz-meta">> = N,V}) ->
    {N, V};
camelize_header({N,V}) when is_binary(N) ->
    {cowboy_bstr:capitalize_token(N), V};
camelize_header({N,V}) ->
    {N, V}.
```

Comment (windkit) [reply src/cowboy_req.erl]: Thanks for the comment, I have updated the PR

**Comment by mocchira** [comment]
LGTM.

@yosukehara finished review and confirmed leo_gateway unit test passed with this PR in my env.
so please merge if 1.3.1 will include this PR.

**Comment by yosukehara** [comment]
@windkit Reviewed and merged, thanks

**Pull request merged**